core/trino-spi/src/main/java/io/trino/spi/block/SingleRowBlockWriter.java

```diff
@@ -24,24 +24,18 @@
         extends AbstractSingleRowBlock
         implements BlockBuilder
 {
-    private static final int INSTANCE_SIZE = ClassLayout.parseClass(SingleRowBlockWriter.class).instanceSize();
+    static final int INSTANCE_SIZE = ClassLayout.parseClass(SingleRowBlockWriter.class).instanceSize();
 
     private final BlockBuilder[] fieldBlockBuilders;
-    private final long initialBlockBuilderSize;
-    private int positionsWritten;
 
     private int currentFieldIndexToWrite;
+    private int rowIndex;
     private boolean fieldBlockBuilderReturned;
 
     SingleRowBlockWriter(int rowIndex, BlockBuilder[] fieldBlockBuilders)
```

Comment:
You can remove `int rowIndex` parameter, set `rowIndex` to `-1` by default

Reply:
ok